### ISSUE TITLE
Simple Example Not Behaving As Expected

### ISSUE THREAD

**Issue by Caglow** [spacetelescope/drizzle#52]
I created a simple example which demonstrates some unexpected behavior. This example does the following:

1. Create a 101x101 image covered in 1s, except for the central pixel (51, 51 in 1-indexed coordinates) which is 10, with values treated as counts per second. The frame has an image scale of 1 arcsec/px and is rotated 45 deg, with (51, 51) corresponding to RA = Dec = 0.

2. Use Drizzle to map the first image onto a new 1001x1001 frame at 0.1 arcsec/px, with north=up, and (501, 501) at RA = Dec = 0.

3. Save the 101x101 image to in.fits, and the 1001x1001 image to out.fits.

Here's the code:

```
from astropy.io import fits
from astropy.wcs import WCS
from drizzle.drizzle import Drizzle
import numpy as np

img_in = np.ones([101, 101])
img_in[50, 50] = 10

w_in = WCS(naxis=2)
w_in.wcs.ctype = ['RA---TAN', 'DEC--TAN']
w_in.wcs.cdelt = [-1/3600, 1/3600]
w_in.wcs.crota = [45, 45]
w_in.wcs.crpix = [51, 51]
w_in.wcs.crval = [0, 0]
w_in.pixel_shape = [101, 101]

f_in = fits.PrimaryHDU(img_in, header=w_in.to_header())
f_in.writeto('in.fits', overwrite=True)

w_out = WCS(naxis=2)
w_out.wcs.ctype = ['RA---TAN', 'DEC--TAN']
w_out.wcs.cdelt = [-0.1/3600, 0.1/3600]
w_out.wcs.crota = [0, 0]
w_out.wcs.crpix = [501, 501]
w_out.wcs.crval = [0, 0]
w_out.pixel_shape = [1001, 1001]

driz = Drizzle(outwcs=w_out)
driz.add_image(insci=img_in, inwcs=w_in, in_units='cps')
f_out = fits.PrimaryHDU(driz.outsci, header=w_out.to_header())
f_out.writeto('out.fits', overwrite=True)
```

The expected output of this example (out.fits) is a 1001x1001 square with surface brightness 0.01 rotated 45 deg (with corners cut off), and a 10x10 square of surface brightness 0.1 also rotated 45 deg centered on (501, 501)---in other words, essentially the original scaled up by a factor of 10 and rotated about the center by 45 deg.

The actual output deviates in two respects:

1. The upper right part of the big square is cut off (zero, as though it's outside the original image, which it's not). A workaround is to set `w_out.pixel_shape = [2002, 2002]`, then crop `driz.outsci` to `driz.outsci[:1001, :1001]`, which gives the full rotated square (minus the corners, as expected). (A more extreme example: if instead of 45 deg, the input image has crota of 180 deg, out.fits is completely blank)

2. The small square is not centered at (501, 501). Instead, (501, 501) is at one edge of the square. (At input crota of 30 deg and 150 deg, it's closer to the center of the small square; at input crota of 90 deg, it's at the corner of the square)

I attached a [zip file containing both the in.fits and out.fits](https://github.com/spacetelescope/drizzle/files/4413721/in_out.zip) that this example produces.

I've tested the example with drizzle 1.13.1 on two different computers (both amd64 architecture):

1. Debian 10, Python 3.7.3, numpy 1.18.1, astropy 3.1.2
2. RHEL 7.6, Python 3.6.5, numpy 1.18.2, astropy 3.0.3

Both appear to produce identical results, as described.

**Comment by jdavies-st** [comment]
Thanks for the report.  Looks like a bug.  May take a while to get to this.

**Comment by mcara** [comment]
The mis-centering error was probably fixed by #83 and incomplete resampled image was fixed by #104 and #110.
